provider/spark.go

```diff
@@ -478,12 +479,47 @@ func (db *DatabricksExecutor) RunSparkJob(args []string, store SparkFileStore) e
 		JobId: jobToRun.JobId,
 	})
 	if err != nil {
-		return fmt.Errorf("the '%v' job failed: %v", jobToRun.JobId, err)
+		errorMessage, err := db.getErrorMessage(jobToRun.JobId)
+		if err != nil {
+			return fmt.Errorf("the '%v' job failed, could not get error message: %v", jobToRun.JobId, err)
+		}
+
+		return fmt.Errorf("the '%v' job failed: %v", jobToRun.JobId, errorMessage)
 	}
 
 	return nil
 }
 
+func (db *DatabricksExecutor) getErrorMessage(jobId int64) (string, error) {
+	ctx := context.Background()
+
+	runRequest := jobs.ListRunsRequest{
+		JobId: jobId,
+	}
+
+	runs, err := db.client.Jobs.ListRunsAll(ctx, runRequest)
+	if err != nil {
+		return "", fmt.Errorf("could not get run id: %v", err)
+	}
+
+	runID := runs[0].RunId
+	request := jobs.GetRunRequest{
+		RunId: runID,
+	}
+
+	// in order to get the status of the run output, we need to
+	// use API v2.0 instead of v2.1. The following code leverages
+	// version 2.0 of the API to get the run output.
+	var runOutput jobs.RunOutput
+	path := "/api/2.0/jobs/runs/get-output"
```

Comment:
Couple of questions here:
1. Why is this? Or I should ask, will the "why" eventually change, and if so, will we have to update `path`?
2. Do we want to move the value of `path` into configuration with a constant default? Is this overkill?